dd-java-agent/instrumentation/jax-rs-annotations/src/main/java/datadog/trace/instrumentation/jaxrs/JaxRsAnnotationsInstrumentation.java

```diff
@@ -155,11 +153,38 @@ public JaxRsAnnotationsInstrumentation() {
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
-        @Advice.Enter final Stack<Scope> scopes, @Advice.Thrown final Throwable throwable) {
+        @Advice.Origin final Method method,
+        @Advice.Enter final Stack<Scope> scopes,
+        @Advice.Thrown final Throwable throwable) {
       if (throwable != null) {
-        final Span span = scopes.peek().span();
-        Tags.ERROR.set(span, true);
-        span.log(Collections.singletonMap(ERROR_OBJECT, throwable));
+        ArrayList<Class> allowedExceptions = new ArrayList<>();
+
+        Class<?> target = method.getDeclaringClass();
+        final TraceSetting classAnnotation = target.getAnnotation(TraceSetting.class);
```

Comment:
This code duplication could be reduced by taking advantage of the fact that `Class` and `Method` both implement [AnnotatedElement](https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AnnotatedElement.html).  In fact, if you can move this logic to the annotation class that would be even better.

Reply:
Good idea, updated.